tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -81,7 +84,22 @@ void start(String[] args) throws IOException {
 
             List<byte[]> payloadByteList = readPayloadFile(payloadFilePath, payloadDelimiter);
 
-            Properties props = readProps(producerProps, producerConfig, transactionalId, transactionsEnabled);
+            Properties props = readProps(producerProps, producerConfig);
```

Comment:
Maybe we can enhance `readProps` to handle those changes? With that we can write UT for `readProps` easily.

Reply:
I refactored all arguments in ProducerPerformance into a new class to facilitate writing more unit tests and improving test coverage.